docs/guides/custom-functions.md

```diff
@@ -0,0 +1,131 @@
+# Custom Functions
+
+If the built-in functions are not enough, Spectral allows you to write and use your own custom functions.
+
+A custom function might be any JS function compliant with [IFunction](https://github.com/stoplightio/spectral/blob/90a0864863fa232bf367a26dace61fd9f93198db/src/types/function.ts#L3#L8) type.
+
+```ts
+export type IFunction<O = any> = (
+  targetValue: any,
+  options: O,
+  paths: IFunctionPaths,
+  otherValues: IFunctionValues,
+) => void | IFunctionResult[];
+```
+
+Custom functions take exactly the same arguments as built-in functions do, so you are more than welcome to take a look at the existing implementation.
+
+The process of creating a function involves 2 steps:
+
+- create a js file inside of a directory called `functions` that should be placed next to your ruleset file
+- create a `functions` array in your ruleset if you haven't done it yet and place a string with the filename without `.js` extension
+
+#### Example:
+
+**functions/abc.js**
+
+```js
+module.exports = (targetVal) => {
+  if (targetVal !== 'abc') {
+    return [
+      {
+        message: 'Value must equal "abc".',
+      },
+    ];
+  }
+};
+```
+
+**my-ruleset.yaml**
+
+```yaml
+functions: [abc]
+rules:
+  my-rule:
+    message: "{{error}}"
+    given: "$.info"
+    then:
+      function: "abc"
+```
+
+Optionally, if you'd like to validate the data that it passed to abc function before the function gets executed, you can provide a JSON Schema.
+You can do it as follows
+
+```yaml
+functions: [[abc, { type: "string" }]] # can be any valid JSONSchema7
+rules:
+  my-rule:
+    message: "{{error}}"
+    given: "$.info"
+    then:
+      function: "abc"
+```
+
+If for some reason, you do not want to place your functions in a directory called `functions`, you can specify a custom directory.
+
+```yaml
+functions: [[abc, { type: "string" }]] # can be any valid JSONSchema7
+functionsDir: "./my-functions" # any path relative to the ruleset file is okay
+rules:
+  my-rule:
+    message: "{{error}}"
+    given: "$.info"
+    then:
+      function: "abc"
+```
+
+Spectral would look for functions in a `my-functions` directory now.
+
+## Inheritance
+
+Core functions can be overridden with custom rulesets, so if you'd like to make your own truthy go ahead. Custom functions are only available in the ruleset which defines them, so loading a foo in one ruleset will not clobber a foo in another ruleset.
```

Comment:
This is confusing. The `truthy` function is defined further up, but I can still use it in my ruleset. Can I not use any of the functions defined in the oas2 ruleset in my own ruleset file?

Reply:
> Can I not use any of the functions defined in the oas2 ruleset in my own ruleset file?

You can use it.

Reply:
I'm not really sure how to explain it any better :thinking: 
As you could have all observed already, writing docs is not my biggest asset :laughing: